packages/modules/workflow-engine-redis/src/utils/workflow-orchestrator-storage.ts

```diff
@@ -416,7 +416,7 @@ export class RedisDistributedTransactionStorage
     })
 
     // Database operations
-    if (hasFinished && !retentionTime && !idempotent) {
+    if (hasFinished && !retentionTime) {
```

Comment:
Q: Does that mean that if it was idempotent without any retention time then it was just saved and not removed? Just want to be sure ahah. Should we apply the same to the inmemory storage as well? Also, does that mean that now it would only be idempotent for as long as the retention time allows it to be otherwise, if it is idempotent without retention time then it is idempotent for as long as the workflow is being executed?

Reply:
Yes, this is it. you answered it already hehe.
it is applied in both, redis and in-memory.